lib/SILGen/SILGenConstructor.cpp

```diff
@@ -1145,7 +1145,16 @@ void SILGenFunction::emitMemberInitializers(DeclContext *dc,
         auto loweredResultTy = getLoweredType(origType, substType);
         auto loweredSubstTy = getLoweredType(substType);
 
-        if (loweredResultTy != loweredSubstTy) {
+        bool isPropertyWrapper = false;
+        if (auto *singleVar = varPattern->getSingleVar()) {
+          isPropertyWrapper = singleVar->getOriginalWrappedProperty();
+        }
+
+        // A converting initialization isn't necessary if the member is
+        // a property wrapper. Though the initial value can have a
+        // reabstractable type, the result of the initialization is
+        // always the property wrapper type, which is never reabstractable.
+        if (!isPropertyWrapper && loweredResultTy != loweredSubstTy) {
```

Comment:
It might be better to put the calculation of loweredResultTy / loweredSubstTy inside an `if (!isPropertyWrapper)` branch, since whether or not they're equal is not relevant at all in the property wrapper case.